src/flint/flint_base/flint_base.pyx

```diff
@@ -10,78 +12,78 @@ cdef class flint_elem:
-#         all polynomial types).
-
-#         If *ascending* is *True*, the monomials are output from low degree to
-#         high, otherwise from high to low.
-#         """
-#         coeffs = [str(c) for c in self]
-#         if not coeffs:
-#             return "0"
-#         s = []
-#         coeffs = enumerate(coeffs)
-#         if not ascending:
-#             coeffs = reversed(list(coeffs))
-#         for i, c in coeffs:
-#             if c == "0":
-#                 continue
-#             else:
-#                 if c.startswith("-") or (" " in c):
-#                     c = "(" + c + ")"
-#                 if i == 0:
-#                     s.append("%s" % c)
-#                 elif i == 1:
-#                     if c == "1":
-#                         s.append("x")
-#                     else:
-#                         s.append("%s*x" % c)
-#                 else:
-#                     if c == "1":
-#                         s.append("x^%s" % i)
-#                     else:
-#                         s.append("%s*x^%s" % (c, i))
-#         return " + ".join(s)
-
-#     def roots(self, **kwargs):
-#         """
-#         Isolates the complex roots of *self*. See :meth:`.acb_poly.roots`
-#         for details.
-#         """
-#         # TODO: 
-#         # To avoid circular imports, we import within the method
-#         from XXX.XXX.acb_poly import acb_poly
-#         return acb_poly(self).roots(**kwargs)
+
+
+cdef class flint_poly(flint_elem):
+    """
+    Base class for polynomials.
+    """
+
+    def __iter__(self):
+        cdef long i, n
+        n = self.length()
+        for i in range(n):
+            yield self[i]
+
+    def coeffs(self):
+        return list(self)
+
+    def str(self, bint ascending=False):
+        """
+        Convert to a human-readable string (generic implementation for
+        all polynomial types).
+
+        If *ascending* is *True*, the monomials are output from low degree to
+        high, otherwise from high to low.
+        """
+        coeffs = [str(c) for c in self]
+        if not coeffs:
+            return "0"
+        s = []
+        coeffs = enumerate(coeffs)
+        if not ascending:
+            coeffs = reversed(list(coeffs))
+        for i, c in coeffs:
+            if c == "0":
+                continue
+            else:
+                if c.startswith("-") or (" " in c):
+                    c = "(" + c + ")"
+                if i == 0:
+                    s.append("%s" % c)
+                elif i == 1:
+                    if c == "1":
+                        s.append("x")
+                    else:
+                        s.append("%s*x" % c)
+                else:
+                    if c == "1":
+                        s.append("x^%s" % i)
+                    else:
+                        s.append("%s*x^%s" % (c, i))
+        return " + ".join(s)
+
+    def roots(self):
+        """
+        Depreciated function.
```

Comment:
This is a very common misspelling that I see a lot. I guess it makes more sense somehow when translated to English from some other languages.

The word is "deprecate" (to express disapproval of) rather than "depreciate" (to diminish in value over time).

Reply:
Thanks for spotting this, spelling isn't my strong point. Will fix now.